src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -96,11 +96,11 @@ public KafkaTopicRepository(final ZooKeeperHolder zkFactory,
         this.circuitBreakers = new ConcurrentHashMap<>();
     }
 
-    public List<String> listTopics() throws TopicRepositoryException {
+    public Set<String> listTopics() throws TopicRepositoryException {
         try {
-            return zkFactory.get()
-                    .getChildren()
-                    .forPath("/brokers/topics");
+            return kafkaFactory.getConsumer()
+                    .listTopics()
```

Comment:
I would not stop merging this PR since it introduces a new relevant feature, but this change is not strictly related and I'm not confident about the performance impact. Is listing from ZK faster or slower than getting new consumer, and fetching all metadata? It could be faster. It could be slower.

Reply:
You are right. To be honest, I went through the code and only found usages in tests. So, besides the already existent bad smell, no problem here.